classes/hooks.php

```diff
@@ -149,25 +159,35 @@ public function check_order_received() {
 	 */
 	public function handle_refund_callback() {
 		try {
-			$refund = Plugin::gateway()->receive_refund_callback();
-			$meta   = $refund->getMerchantMetadata();
+			$refund     = Plugin::gateway()->receive_refund_callback();
+			$meta       = $refund->getMerchantMetadata();
+			$payment_id = $refund->getPaymentId();
 
 			ZCO()->logger()->debug( 'Received Zaver refund callback', (array) $refund );
 
-			if ( ! isset( $meta['orderId'] ) ) {
+			$order_id = Helper::get_order_by_payment_id( $payment_id );
+			if ( empty( $order_id ) ) {
+				$order_id = $meta['orderId'] ?? null;
+			}
+
+			if ( empty( $order_id ) ) {
 				throw new Exception( 'Missing order ID' );
 			}
 
-			$order = wc_get_order( $meta['orderId'] );
+			$order = wc_get_order( $order_id );
 
 			if ( ! $order ) {
 				throw new Exception( 'Order not found' );
 			}
 
 			Refund_Processor::handle_response( $order, $refund );
 		} catch ( Exception $e ) {
-			// translators: %s is the error message.
-			$order->update_status( 'failed', sprintf( __( 'Failed with Zaver payment: %s', 'zco' ), $e->getMessage() ) );
+
+			// Do not change the status if the order has already been paid.
+			if ( empty( $order->get_date_paid() ) ) {
```

Comment:
Issue: Exceptions are thrown if we fail to get a order id or an order from WooCommerce, which means the $order variable will either be unset or false, meaning that `$order->get_date_paid` and `$order->update_stauts` will generate errors.

Reply:
Yes, i don't think there is much more we can do in that case sadly since we cant locate an order for it.